compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt

```diff
@@ -65,6 +77,36 @@ internal class PlatformWindowContext {
         )
     }
 
+    /**
+     * Converts the given [Rect] from the coordinate space of the container window to the coordinate
+     * space of the root UIWindow in which hierarchy with the container window resides.
+     */
+    fun convertContainerWindowRectToRootWindowCGRect(rect: Rect): CValue<CGRect> {
+        val windowContainer = _windowContainer ?: return CGRectZero.readValue()
+
+        val windowContainerWindow = windowContainer.window
+
+        fun UIWindow.density(): Density {
+            return Density(screen.scale.toFloat())
+        }
+
+        return if (windowContainerWindow == null) {
+            if (windowContainer is UIWindow) {
+                // windowContainer is the root window itself
+                rect.toDpRect(windowContainer.density()).asCGRect()
+            } else {
+                // windowContainer is not attached to any window
+                CGRectZero.readValue()
+            }
+        } else {
```

Comment:
check if `windowContainerWindow != windowContainer`

Reply:
windowContainer.window will be null in case it's true, it's a redundant check.